flight/Libraries/math/pid.c

```diff
@@ -199,6 +230,67 @@ void pid_configure(struct pid *pid, float p, float i, float d, float iLim)
 }
 
 /**
+ * Assigns a pid_deadband struct to the PID controller.
+ * @param[out] pid The PID structure to configure
+ * @param[in] deadband The external deadband structure to assign
+ */
+void pid_assign_deadband(struct pid *pid, struct pid_deadband *db)
+{
```

Comment:
> I don't understand this-- if we have a deadband on the rate controller, shouldn't it always be there-- whether we're flying rate or attitude?

Say someone is switching between axis modes with and without deadband during uptime, seems less drama to allocate it once and use it conditionally, than free and reallocate it every time. The code for this is simpler. For instance, I've provisionally excluded weak leveling because of its slow acceleration. Or someone may decide to run autotune and then decide to run down the battery in attitude. SettingsCb would just malloc the structure and it then gets used when needed or NULL gets passed in the respective axis modes. For instance, axis lock might possibly be another for exclusion, too (then again, I think you wanted it gone, anyway).